File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/TiledMapAnimationHandler.java
/**
 * 
 */
package com.abubusoft.xenon.animations;

import com.abubusoft.xenon.animations.events.EventFrameListener;

/**
 * Rappresenta l'handler delle animazioni su tiledmap che comprendono sia lo spostamento che l'animazione di uno sprite.
 * 
 * @author Francesco Benincasa
 *
 */
public class TiledMapAnimationHandler extends Parallel2Handler<TranslationFrame, TextureKeyFrame> {

	public TiledMapAnimationHandler()
	{
		handler0=new TranslationHandler();
		handler1=new TextureAnimationHandler();
	}
	

	/**
	 * Imposta il listener relativo allo spostamento.
	 * @param value
	 */
	public void setOnMoveEventListener(EventFrameListener<TranslationFrame> value) {
		handler0.setFrameListener(value);
	}
	
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/TiledMapAnimationHandler.kt
/**
 *
 */
package com.abubusoft.xenon.animations

import com.abubusoft.xenon.animations.events.EventFrameListener

/**
 * Rappresenta l'handler delle animazioni su tiledmap che comprendono sia lo spostamento che l'animazione di uno sprite.
 *
 * @author Francesco Benincasa
 */
class TiledMapAnimationHandler : Parallel2Handler<TranslationFrame?, TextureKeyFrame?>() {
    init {
        handler0 = TranslationHandler()
        handler1 = TextureAnimationHandler()
    }

    /**
     * Imposta il listener relativo allo spostamento.
     * @param value
     */
    fun setOnMoveEventListener(value: EventFrameListener<TranslationFrame?>?) {
        handler0!!.setFrameListener(value)
    }
}